### ISSUE TITLE
UpdateManager crashes during ApplyReleases (Path cannot be empty string or all whitespace.)

### ISSUE THREAD

**Issue by ezmani** [Squirrel/Squirrel.Windows#1536]
Hi there!

We are currently using Squirrel version 1.9.0 and splat version 1.6.2. 

Our application periodically checks for update using the update manager, downloads the releases and then uses the ApplyReleases method to complete the process. This worked pretty well in the past.

We ran into a situation where the ApplyReleases is creating the new version folder and then crashed with the following error. 

>  **Path cannot be the empty string or all whitespace.**

   ```

at System.IO.Directory.CreateDirectory(String path)
   at Squirrel.UpdateManager.ApplyReleasesImpl.getLinkTarget(ShortcutLocation location, String title, String applicationName, Boolean createDirectoryIfNecessary)
   at Squirrel.UpdateManager.ApplyReleasesImpl.linkTargetForVersionInfo(ShortcutLocation location, IPackage package, FileVersionInfo versionInfo)
   at Squirrel.UpdateManager.ApplyReleasesImpl.CreateShortcutsForExecutable(String exeName, ShortcutLocation locations, Boolean updateOnly, String programArguments, String icon)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass10_0.<invokePostInstall>b__4(String x)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<invokePostInstall>d__10.MoveNext()
```

Unfortunately due to the crash, it leaves behind the folders older than the current version and the previous version. Has anyone seen this error before?

Any help would be much appreciated!
Mani

**Comment by anaisbetts** [comment]
I would try to debug and see why the path passed along to CreateDriectory is all blank

**Comment by ezmani** [comment]
Hello there! I attempted to debug on my local machine and the path is not empty and sure enough works as expected. I was able to enable squirrel logging and attached it with this post.

The application in production with Squirrel runs on computer with much less privileges. I see it failing to create a directory for the desktop location.

**Is there anyway to disable the shortcut creation?**


[SquirrelInternal.log](https://github.com/Squirrel/Squirrel.Windows/files/3569159/SquirrelInternal.log)
